### ISSUE TITLE
Why we are not shuffle real and fake images in batch?

### ISSUE THREAD

**Issue by mrgloom** [eriklindernoren/Keras-GAN#146]
Why we are not shuffle real and fake images in batch?

For example here:
https://github.com/eriklindernoren/Keras-GAN/blob/master/cgan/cgan.py#L140

**Comment by nobodykid** [comment]
I agree, most GAN implementations I see does various sort of shuffle to maintain randomness in sampling image when training model

**Comment by mrgloom** [comment]
Maybe it's related to https://github.com/soumith/ganhacks#4-batchnorm ? But looks like it's only related when BN layers are used.

**Comment by mrgloom** [comment]
Looks related https://github.com/eriklindernoren/Keras-GAN/issues/9#issuecomment-377700670

**Comment by nobodykid** [comment]
That's all on BatchNorm, while the pix2pix in this repo already changed it with Instance Norm

**Comment by mrgloom** [comment]
As I can see here 
https://github.com/eriklindernoren/Keras-GAN/blob/master/pix2pix/pix2pix.py#L83
BN is used

**Comment by nobodykid** [comment]
I see, I forgot that was my version of pix2pix where I switch BN with IN
Pardon about that

**Comment by mrgloom** [comment]
Also here is a note in section 4.3.1
https://cs.nyu.edu/media/publications/mathieu_michael.pdf
"we observed that Batch  Normalization  works  better  when  the  minibatches  are  separated  between real and fake samples, which may not be intuitive."

**Comment by eriklindernoren** [comment]
I have made this decision based on the reasons already discussed in this issue, i.e. for BN first and foremost. For the other implementations where BN is not used I haven't really found any benefits of mixing real and fake samples.

**Comment by mrgloom** [comment]
Also related to BN
https://github.com/xinntao/ESRGAN#bn-artifacts

Also seems BN is not common to use nowadays, for example adain is used in stylegan.
